### DISCUSSION TITLE
Magical Disappearing Backups?

### DISCUSSION THREAD

**Discussion by Majoraslayer** [nextcloud/all-in-one#4814] [Questions] [answered]
I'm trying to migrate my server to a new OS install, and copied over all of my docker run configs. The new install has all of the same mount points for storage.  Keep in mind, I'm doing this with another OS storage drive, so I can swap back and forth between the old and new installs by swapping out the drive.

To migrate my Nextcloud AIO instance over, I ran the backup on the old install AND ran the integrity check on the backup.  Everything came out fine.  When I swapped the drive out and opened up the new AIO interface, I tried to restore the backup (mounted in the same place) only to find the backup no longer exists.  The borg folder is where it should be, but there's nothing in it.  I've tried this twice.  The only thing I know to do is try to copy the backup somewhere else so I can restore it, but I haven't had time to try it yet due to my work schedule.

I have no idea how or why it's doing this.  Is the AIO somehow reading the location of the backup from the mounted config folder, clearing all backups, then refusing to acknowledge the existing config?  I can't figure out what I'm missing or what could possibly be deleting these backups.

**Comment by Majoraslayer** [comment]
Alright, so I dug into it deeper, and while it's generating the borg folder at the destination I configured, it's not actually putting anything in it.  This makes absolutely no sense because it SAYS the backup was created successfully with no errors.  I can even run integrity checks on the backup, but IT. IS. NOT. THERE.  It should be going to a volume I mounted at /drives/SharedDrive/Backups/Nextcloud.  I can use docker exec -it container_name bash to get into the shell inside the AIO container, navigate to the configured backup location, and I don't even see it from there.  I have no idea where Nextcloud is seeing the backups it's running integrity checks on because it's definitely not in the configured folder.

I eventually gave up on migrating my old install, so I just set up the new install from scratch.  Then I configured the backups again, this time in a new location......AND IT'S DOING THE SAME THING.

For additional detail, my OS is Debian.  I've seen mention of a similar problem under Windows, but my new install is on Debian 12 while my old one was on Ubuntu 24.03.

**Comment by szaimen** [comment, marked as answer]
Hi, can you post the output of `sudo docker inspect nextcloud-aio-borgbackup` here?

**Reply by Majoraslayer** [reply]
This may be a dumb question, but do I need to try to run the backup service again first to get it to create that container?  I don't see nextcloud-aio-borgbackup in my "docker ps" list, and the output confirms it with "No such container"

**Reply by szaimen** [reply]
Yeah I guess so if you deleted it in the meantime

**Reply by Majoraslayer** [reply]
[borgbackup-log.txt](https://github.com/user-attachments/files/15812018/borgbackup-log.txt)
My output should be attached.  I don't remember explicitly killing the container, but nevertheless running a backup seems to have created it now.

**Reply by szaimen** [reply]
>               "Source": "/drives/SharedDrive/Nextcloud/Backup",
 >               "Destination": "/mnt/borgbackup",

i see. And you say the backup archive is not created in /drives/SharedDrive/Nextcloud/Backup/borg on the host?

**Reply by Majoraslayer** [reply]
Correct, but I believe I finally solved the mystery.  

I didn't realize that the backup process **runs in another container separate from the master**.  My external drives are all mounted under /mnt on my host system, but since the docker run command runs the master container's data within /mnt/docker-aio-config, I didn't want that to cause a conflict by mapping those drives through /mnt.  So, as a workaround, I mapped /mnt:/drives.  Now that I know that the backups run from a separate container, I realized the volume mapping wouldn't carry over to separate containers created by the Master container.  I just created a softlink in the root of my host system at /drives pointing to /mnt, and now the backups seem to be created as intended.

Thanks for the help!
